File: testserver/src/RoomManager.js
import Room from "./Room";

export default class RoomManager {
    rooms;
    lobby;
    server;

    constructor(server) {
        this.server = server;
        this.rooms = new Map();
        this.lobby = new Room(this, "Lobby");
        this.lobby.temp = false;
        this.add(this.lobby);
    }

    remove(room) {
        this.rooms.delete(room.name);
        this.server.broadcast({roomClosed: {name: room.name}});
    }

    add(room) {
        this.rooms.set(room.name, room);
        this.server.broadcast({roomCreated: {name: room.name}});
    }

    get(name) {
        let room;
        if(room = this.rooms.get(name)) {
            return room;
        }
        room = new Room(this, name);
        this.add(room);
        return room;
    }
}

File: server/src/server/RoomManager.java
package server;

import java.util.HashMap;
import java.util.Map;

public class RoomManager {
	
	private ChatWebSocketHandler server;
	public Map<String, Room> rooms;
	Room lobby;
	
	 public RoomManager(ChatWebSocketHandler server) {
	        this.server = server;
	        this.rooms = new HashMap<String, Room>();
	        this.lobby = new Room(this, "Lobby");
	        this.lobby.temp = false;
	        this.add(this.lobby);
	    }

	    public void remove(Room room) {
	        this.rooms.remove(room.name);
	        this.server.broadcastRoomClosed(room.name);
	    }

	    public void add(Room room) {
	        this.rooms.put(room.name, room);
	        this.server.broadcastRoomCreated(room.name);
	    }

	    public Room get(String name) {
	        Room room = this.rooms.get(name);
	        if(room!=null) {
	            return room;
	        }
	        room = new Room(this, name);
	        this.add(room);
	        return room;
	    }
	    
	    

}






   
